---
title: "RNA-seq analysis in R"
subtitle: "Differential Expression of RNA-seq data"
author: "Stephane Ballereau, Dominique-Laurent Couturier, Mark Dunning, Abbi Edwards, Ashley Sawle"
date: '`r format(Sys.time(), "Last modified: %d %b %Y")`'
output:  html_document
---

```{r setup, echo=FALSE}
options(tibble.print_max = 4, tibble.print_min = 4, max.print=40, 
        tibble.max_extra_cols=2)
```

```{r echo=FALSE, message = FALSE}
library(tidyverse)
library(DESeq2)

# load the RData object we created in the previous session
load("../Robjects/preprocessing.RData")
```

### Exercise 1
>
> Plot the biased and unbiased MA plots both samples side by side to see the 
> before and after normalisation.
>

```{r}
# recap
# Read the sample information into R
sampleinfo <- read.delim("../data/SampleInfo.Corrected.txt", 
                         stringsAsFactors = F)
# Read the data into R
seqdata <- read.delim("../data/GSE60450_Lactation.featureCounts", 
                      stringsAsFactors = F,
                      comment = "#")
# Remove first two columns from seqdata
countdata <- as.data.frame(seqdata) %>% 
    column_to_rownames("Geneid") %>% # turn the geneid column into rownames
    rename_all(str_remove, ".bam") %>% # remove the ".bam" from the column names
    select(sampleinfo$Sample) %>% # keep sample columns using sampleinfo
    as.matrix()
# filter data
keep <- rowSums(countdata) > 5
countdata <- countdata[keep,]
# rlogcounts
rlogcounts <- rlog(countdata)
# DGE list
design <- as.formula(~ CellType)
# create the DESeqDataSet object
ddsObj <- DESeqDataSetFromMatrix(countData = countdata,
                              colData = sampleinfo,
                              design = design)
ddsObj <- estimateSizeFactors(ddsObj)
# normalised counts
logcounts <- log2(countdata + 1)
normalizedCounts <- counts(ddsObj, normalized=TRUE) 
logNormalizedCounts <- log2(normalizedCounts + 1)
```


```{r solutionChallenge3, fig.height=10, fig.width=10, collapse=TRUE}
library(limma)
par(mfrow=c(2,2))
plotMA(logcounts, array = 7, main="MCL1.LA - Raw")
abline(h=0,col="grey")
plotMA(logNormalizedCounts, array = 7, main="MCL1.LA - Normalised")
abline(h=0,col="grey")
plotMA(logcounts, array = 11, main="MCL1.LE - Raw")
abline(h=0,col="grey")
plotMA(logNormalizedCounts, array = 11, main="MCL1.LE - Normalised")
abline(h=0,col="grey")
```


### Exercise 2

First we will recapitulate the above steps to generate a new DESeq2 object with
the additive model. Then we will extract the results table for the contrast
for "lactate v virgin".

```{r}
design <- as.formula(~ CellType + Status)
ddsObj.raw <- DESeqDataSetFromMatrix(countData = countdata, 
                                     colData = sampleinfo,
                                     design = design)
ddsObj <- DESeq(ddsObj.raw)
resLvV <- results(ddsObj, alpha=0.05)
```

#### **Challenge 1** {.challenge}

a) Obtain results for luminal vs basal.  Call the new results object `resLvB`.

```{r}
resultsNames(ddsObj)
resBvL <- results(ddsObj, alpha=0.05, name="CellType_luminal_vs_basal")
resBvL
```


b) How many significantly upregulated genes are there?

```{r}
sum(resBvL$padj < 0.05 & resBvL$log2FoldChange > 0, na.rm = TRUE)
```

#### **Challenge 2 - Contrasts** {.challenge} 

Suppose we want to find genes that are  differentially expressed between 
**pregnant** and **lactate**. We don't have a parameter that explicitly will 
allow us to test that hypothesis. We need to provide a contrast.
Look at the help page for `results` (`?results`) and read about the `contrast`
argument (also look at Example 1 at the bottom of the help page)

Use the `contrast` argument to extract the results table for "pregnant v
lactate".

```{r}
resPvL <- results(ddsObj, 
                  alpha=0.05,
                  contrast = c("Status", "pregnant", "lactate"))
resPvL
```

### Exercise 3

When we looked at the PCA it did seem that an interaction model might be
warranted. Let's test that.  

1. Create a new DESeq2 object using a model with an interaction between 
   CellType and Status (`~ CellType * Status`)

```{r}
design <- as.formula(~ CellType * Status)
ddsObj2.raw <- DESeqDataSetFromMatrix(countData = countdata, 
                                     colData = sampleinfo,
                                     design = design)
```

2. Use the LRT to compare this to the simpler additive model 
  (`~CellType + Status`)

```{r}
# create the simpler model
design.reduced <- as.formula(~ CellType + Status)

# Compare the two designs
ddsObjC2 <- DESeq(ddsObj2.raw, test="LRT", reduced=design.reduced)
resCSvCxS <- results(ddsObjC2)
```

3. For how many genes is interaction model a better fit?  

```{r}
sum(resCSvCxS$padj < 0.05, na.rm=TRUE)
```

--------------------

