---
title: "RNA-seq Analysis in R"
subtitle: "Annotation and Visualisation of RNA-seq results - Solutions"
author: "Stephane Ballereau, Dominique-Laurent Couturier, Mark Dunning, Abbi Edwards, Ashley Sawle"
date: '`r format(Sys.time(), "Last modified: %d %b %Y")`'
output:
  html_document:
    toc: yes
    toc_float: yes
minutes: 300
layout: page
---

```{r setup, include=FALSE}
library(EnsDb.Mmusculus.v79)
library(GenomicFeatures)
library(DESeq2)
library(tidyverse)
```

```{r prepareData, echo=FALSE, message=FALSE, warning=FALSE}
load("Robjects/DE.RData")
```

### Challenge 1

## Retrieve full annotation

> ### Challenge 1 {.challenge}
> That was just 1000 genes. We need annotations for the entire results table.
>   
> (a) Query BioMart using all of the genes in our results table (`resLvV`)  
>
> (b) Can we also have the biotype of our genes too?
>
> (c) How many Ensembl genes have multipe Entrez IDs associated with them?  


```{r solutionChallenge1code, eval=FALSE, message=FALSE, warnings=FALSE}

# lets set it up
ourCols <- c("SYMBOL", "GENEID", "ENTREZID", "GENEBIOTYPE")
ourKeys <- rownames(resLvV)

# run the query
annot <- select(EnsDb.Mmusculus.v79, 
                keys=ourKeys, 
                columns=ourCols, 
                keytype="GENEID")
# multiple EntrezIDs
annot %>%  
    add_count(GENEID) %>%  
    dplyr::filter(n>1)
```

### Challenge 2 {.challenge}

Use the log2 fold change (`logFC`) on the x-axis, and use `-log10(FDR)` on the
y-axis. (This >`-log10` transformation is commonly used for p-values as it
means that more significant genes have a higher scale).

> (a) Create a column of -log10(FDR) values
>
> (b) Create a plot with points coloured by if FDR < 0.05


```{r volcanoPlot, fig.height=5, fig.width=7}
# first remove the filtered genes (FDR=NA) and create a -log10(FDR) column
filtTab <- shrinkLvV %>% 
    filter(!is.na(FDR)) %>% 
    mutate(`-log10(FDR)` = -log10(FDR))

ggplot(filtTab, aes(x = logFC, y=`-log10(FDR)`)) + 
    geom_point(aes(colour=FDR < 0.05), size=1)
```


### Challenge 3 - In Supplementary Materials

Use the txMm to retrieve the exon coordinates for the genes: 

> * `ENSMUSG00000021604`  
> * `ENSMUSG00000022146`  
> * `ENSMUSG00000040118`   


```{r solutionChallenge3, warning=FALSE, message=FALSE}
keyList <- c("ENSMUSG00000021604", "ENSMUSG00000022146", "ENSMUSG00000040118")
AnnotationDbi::select(txMm, 
       keys=keyList,
       keytype = "GENEID",
       columns=c("TXNAME", "TXCHROM", "TXSTART", "TXEND", "TXSTRAND", "TXTYPE")
      )
```
